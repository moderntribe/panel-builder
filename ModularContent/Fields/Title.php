<?php

namespace ModularContent\Fields;
use ModularContent\Panel;

/**
 * Class Title
 *
 * @package ModularContent\Fields
 *
 * A special case of the text field. The title
 * field is automatically added to each Panel,
 * and is used to set the title of the panel
 * in the admin.
 */
class Title extends Text {
	protected $label = '';
	protected $name = 'title';

	public function __construct( $args = array() ){
		$this->label = __( 'Title', 'modular-content' );
		parent::__construct($args);
	}
}
