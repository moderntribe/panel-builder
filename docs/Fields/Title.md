## Title

A special case of the text field. The title field is automatically added to each Panel, and is used to set the title of the panel in the admin.

### Parameters

None - field is automatically added.