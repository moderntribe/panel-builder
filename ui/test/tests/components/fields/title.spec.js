/* global expect b:true */

import React from 'react';
import { shallow, mount } from 'enzyme';
import Title from 'components/fields/title';
import renderer from 'react-test-renderer';

describe('Title field', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<Title description="This is a test description" label="Some test label" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders as a <div>', () => {
		const wrapper = shallow(<Title />);
		expect(wrapper.type()).toEqual('div');
	});

	it('allows us to set props', () => {
		const wrapper = mount(<Title description="This is a test" />);
		expect(wrapper.props().description).toEqual('This is a test');
		wrapper.setProps({ description: 'Change props for test' });
		expect(wrapper.props().description).toEqual('Change props for test');
	});

	it('has a label', () => {
		const wrapper = shallow(<Title />);
		expect(wrapper.find('label').length).toEqual(1);
	});

	it('renders a text input', () => {
		const wrapper = shallow(<Title />);
		expect(wrapper.find('input[type="text"]').length).toEqual(1);
	});
});
