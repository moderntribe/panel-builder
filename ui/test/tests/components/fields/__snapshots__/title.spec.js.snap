exports[`Title field renders correctly 1`] = `
<div
  className="field panel-field">
  <label
    className="label panel-input-label-title panel-field-label">
    Some test label
    <span
      className="icon"
      onBlur={undefined}
      onClick={undefined}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined} />
  </label>
  <span
    className="inputContainer">
    <input
      name="modular-content-"
      onChange={[Function]}
      type="text"
      value="" />
  </span>
</div>
`;
